### PR TITLE
fix: [Issue #405] Add scroll event to close popup panels

### PR DESCRIPTION
**Pull request by ajaguasin**
Hi, this is my first time contributing.

I wasn't sure where to add my username and the date for each file I changed.

Please let me know, and i'll adjust accordingly,

Thanks!

### CODE REVIEW

File: _assets/js/main.js [hunk 1]
```diff
@@ -875,6 +875,10 @@ jQuery(document).ready(function($) {
             if (e.key == 'Escape' && $panel.is(':visible')) {
               closePanel($link, $panel);
             }
+          })
+          .on('scroll', function() {
```
Comment (muffinresearch) [inline _assets/js/main.js]: This event should be namespaced - see above for examples. It should also not operate if there's no panel open.

File: _assets/js/main.js [hunk 2]
```diff
@@ -875,6 +875,10 @@ jQuery(document).ready(function($) {
...
+            closePanel($link, $panel);
+            $(document).off('scroll');
```
Comment (muffinresearch) [inline _assets/js/main.js]: This should probably be called in `closePanel` instead (don't forget to use the namespace).

**Comment by muffinresearch** [comment]
On reflection, I'm wondering if scroll is the right way to do this, really this should fire when the digram is no longer in-view. Since the diagram's animations are tied to being in view, that might provide a better hook rather than simply listening to scroll. The problem with scroll is that as soon as you touch the mouse the overlay will be closed, that may create an accessibility issue.

File: _assets/js/main.js (hunk 1 above)
Comment (muffinresearch) [reply _assets/js/main.js]: See my latest comment, I think scroll events might not be the best hook for this.

**Comment by ajaguasin** [comment]
@muffinresearch Thanks for the feedback. I'll investigate it more and update the PR

**Comment by muffinresearch** [comment]
See also https://github.com/mozilla/extension-workshop/issues/405#issuecomment-540973143

**Comment by muffinresearch** [comment]
@ajaguasin hi, since this hasn't seen updates for a while I'm going to close this, please feel free to re-open it when it's ready for another look.

**Pull request closed without merging**